plugins/backend/catalog-backend-module-okta/src/providers/OktaEntityProvider.ts

```diff
@@ -42,17 +42,43 @@ export abstract class OktaEntityProvider implements EntityProvider {
     this.logger = options.logger;
   }
 
-  protected getClient(orgUrl: string): Client {
+  protected getClient(
+    orgUrl: string,
+    oauthScopes: string[] | undefined = undefined,
+  ): Client {
     const account = this.accounts.find(
       acccountConfig => acccountConfig.orgUrl === orgUrl,
     );
     if (!account) {
       throw new Error(`accountConfig for ${orgUrl} not found`);
     }
-    return new Client({
-      orgUrl: account.orgUrl,
-      token: account.token,
-    });
+
+    if (
+      account.oauth &&
+      account.oauth.keyId &&
```

Comment:
KeyId is only required if the privateKey format is a PEM *and*  client app uses multiple JWKs. The conditional requires that keyId will also be set, even if passing a JSON object in JWK format with the kid field set or using a single JWK with the app.

Reply:
Ah yep this makes more sense now i was scratching my head about that. I think I'll just make it optional and defer to the okta sdk to throw if it's missing in the PEM scenario :)

Reply:
fixed